src/PathContainer.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Dhii\Container;
+
+use Psr\Container\ContainerInterface;
+use function array_filter;
+use function ltrim;
+
+/**
+ * A container implementation that wraps around another to provide access to its values via partial path-like keys.
+ *
+ * @since [*next-version*]
+ */
+class PathContainer implements ContainerInterface
```

Comment:
Had to read the test to understand what this does, so a couple of usage examples would be great!